sql-bricks.js

```diff
@@ -53,11 +53,10 @@
 
     // shift the placeholder indexes if there are already values
     if (opts.value_ix > 1) {
-      var orig_val_ix = opts.value_ix;
       if (opts.placeholder == '$%d')
-        str = str.replace(/\$(\d+)/g, function(match, capture) { opts.value_ix++; return '$' + (parseInt(capture, 10) + orig_val_ix - 1); });
+        str = str.replace(/\$(\d+)/g, function(match, capture) { return '$' + (parseInt(capture, 10) + opts.value_ix - 1); });
       else if (opts.placeholder == '?%d')
-        str = str.replace(/\?(\d+)/g, function(match, capture) { opts.value_ix++; return '?' + parseInt(capture, 10) + orig_val_ix - 1; });
+        str = str.replace(/\?(\d+)/g, function(match, capture) { return '?' + parseInt(capture, 10) + opts.value_ix - 1; });

```

Comment:
@vitalif: At first I didn't think this could work, but I tested it and it does work:

```javascript
> select().where(sql('first = $1 AND last = $2', 'Fred', 'Flintstone'), {name: 'Fred Flintstone'}).toParams()
{ text: 'SELECT * WHERE first = $1 AND last = $2 AND name = $3',
  values: [ 'Fred', 'Flintstone', 'Fred Flintstone' ] }
```

Then I realized it was the line below, `opts.value_ix += this.vals.length` that takes the place of the `opts.value_ix++`. To be honest, I'm not sure if that (relying on the number of values passed) is better than the previous approach (relying on the number of replaced expressions in the string), but there may be side-effects I'm not seeing or not thinking about.